[llvm] build failure

**Host Environment**
- OS: Windows
- Compiler: MSVC 19.25.28614

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install llvm[core,tools,utils,target-x86]:x64-windows`

**Failure logs** 
```
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:116 (message):
  'E:/Tools/vcpkg/packages/llvm_x64-windows/debug/share/clang' does not
  exist.
```
[stdout-x64-windows.log](https://github.com/microsoft/vcpkg/files/4575216/stdout-x64-windows.log)

**Additional context**
Fails when building without `clang` feature. Likely cause:
https://github.com/microsoft/vcpkg/blob/cfc05991f5a73ee4b2df7d9132fc8a5a6a42224e/ports/llvm/portfile.cmake#L169

--- a/ports/llvm/CONTROL
+++ b/ports/llvm/CONTROL
@@ -1,5 +1,5 @@
 Source: llvm
-Version: 10.0.0
+Version: 10.0.0-1
 Homepage: https://llvm.org/
 Description: The LLVM Compiler Infrastructure
 Supports: !uwp
@@ -17,14 +17,38 @@ Description: Build with all backends.
 Feature: target-aarch64
 Description: Build with AArch64 backend.
 
+Feature: target-amdgpu
+Description: Build with AMDGPU backend.
+
 Feature: target-arm
 Description: Build with ARM backend.
+
+Feature: target-bpf
+Description: Build with BPF backend.
+
+Feature: target-hexagon
+Description: Build with Hexagon backend.
+
+Feature: target-lanai
+Description: Build with Lanai backend.
 
 Feature: target-mips
 Description: Build with Mips backend.
 
+Feature: target-msp430
+Description: Build with MSP430 backend.
+
+Feature: target-nvptx
+Description: Build with NVPTX backend.
+
 Feature: target-powerpc
 Description: Build with PowerPC backend.
+
+Feature: target-riscv
+Description: Build with RISCV backend.
+
+Feature: target-sparc
+Description: Build with Sparc backend.
 
 Feature: target-systemz
 Description: Build with SystemZ backend.
@@ -34,6 +58,9 @@ Description: Build with WebAssembly backend.
 
 Feature: target-x86
 Description: Build with X86 backend.
+
+Feature: target-xcore
+Description: Build with XCore backend.
 
 Feature: enable-rtti
 Description: Build LLVM with run-time type information.

--- a/ports/llvm/CONTROL
+++ b/ports/llvm/CONTROL
@@ -1,5 +1,5 @@
 Source: llvm
-Version: 10.0.0-1
+Version: 10.0.0-2
 Homepage: https://llvm.org/
 Description: The LLVM Compiler Infrastructure
 Supports: !uwp

--- a/ports/llvm/portfile.cmake
+++ b/ports/llvm/portfile.cmake
@@ -61,6 +61,12 @@ if("clang" IN_LIST FEATURES OR "clang-tools-extra" IN_LIST FEATURES)
             -DCLANG_ENABLE_STATIC_ANALYZER=OFF
         )
     endif()
+    if(VCPKG_TARGET_IS_WINDOWS)
+        list(APPEND FEATURE_OPTIONS
+            # Disable dl library on Windows
+            -DDL_LIBRARY_PATH:FILEPATH=
+        )
+    endif()
 endif()
 if("clang-tools-extra" IN_LIST FEATURES)
     list(APPEND LLVM_ENABLE_PROJECTS "clang-tools-extra")

--- a/ports/llvm/portfile.cmake
+++ b/ports/llvm/portfile.cmake
@@ -93,14 +93,38 @@ else()
     if("target-aarch64" IN_LIST FEATURES)
         list(APPEND LLVM_TARGETS_TO_BUILD "AArch64")
     endif()
+    if("target-amdgpu" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "AMDGPU")
+    endif()
     if("target-arm" IN_LIST FEATURES)
         list(APPEND LLVM_TARGETS_TO_BUILD "ARM")
+    endif()
+    if("target-bpf" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "BPF")
+    endif()
+    if("target-hexagon" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "Hexagon")
+    endif()
+    if("target-lanai" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "Lanai")
     endif()
     if("target-mips" IN_LIST FEATURES)
         list(APPEND LLVM_TARGETS_TO_BUILD "Mips")
     endif()
+    if("target-msp430" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "MSP430")
+    endif()
+    if("target-nvptx" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "NVPTX")
+    endif()
     if("target-powerpc" IN_LIST FEATURES)
         list(APPEND LLVM_TARGETS_TO_BUILD "PowerPC")
+    endif()
+    if("target-riscv" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "RISCV")
+    endif()
+    if("target-sparc" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "Sparc")
     endif()
     if("target-systemz" IN_LIST FEATURES)
         list(APPEND LLVM_TARGETS_TO_BUILD "SystemZ")
@@ -110,6 +134,9 @@ else()
     endif()
     if("target-x86" IN_LIST FEATURES)
         list(APPEND LLVM_TARGETS_TO_BUILD "X86")
+    endif()
+    if("target-xcore" IN_LIST FEATURES)
+        list(APPEND LLVM_TARGETS_TO_BUILD "XCore")
     endif()
 endif()
 
@@ -126,7 +153,7 @@ if("${LLVM_TARGETS_TO_BUILD}" STREQUAL "")
     endif()
 endif()
 
-# Use comma-separated string for enabled projects instead of semicolon-separated string.
+# Use comma-separated string instead of semicolon-separated string.
 # See https://github.com/microsoft/vcpkg/issues/4320
 string(REPLACE ";" "," LLVM_ENABLE_PROJECTS "${LLVM_ENABLE_PROJECTS}")
 string(REPLACE ";" "," LLVM_TARGETS_TO_BUILD "${LLVM_TARGETS_TO_BUILD}")
@@ -164,11 +191,16 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH share/llvm)
 vcpkg_fixup_cmake_targets(CONFIG_PATH share/clang TARGET_PATH share/clang)
 
 if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
-    file(GLOB_RECURSE _release_targets
+    file(GLOB_RECURSE _llvm_release_targets
         "${CURRENT_PACKAGES_DIR}/share/llvm/*-release.cmake"
-        "${CURRENT_PACKAGES_DIR}/share/clang/*-release.cmake"
     )
-    foreach(_target IN LISTS _release_targets)
+    set(_clang_release_targets)
+    if("clang" IN_LIST FEATURES)
+        file(GLOB_RECURSE _clang_release_targets
+            "${CURRENT_PACKAGES_DIR}/share/clang/*-release.cmake"
+        )
+    endif()
+    foreach(_target IN LISTS _llvm_release_targets _clang_release_targets)
         file(READ ${_target} _contents)
         # LLVM tools should be located in the bin folder because llvm-config expects to be inside a bin dir.
         # Rename `/tools/${PORT}` to `/bin` back because there is no way to avoid this in vcpkg_fixup_cmake_targets.
@@ -178,11 +210,17 @@ if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
 endif()
 
 if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
-    file(GLOB_RECURSE _debug_targets
+    file(GLOB_RECURSE _llvm_debug_targets
         "${CURRENT_PACKAGES_DIR}/share/llvm/*-debug.cmake"
         "${CURRENT_PACKAGES_DIR}/share/clang/*-debug.cmake"
     )
-    foreach(_target IN LISTS _debug_targets)
+    set(_clang_debug_targets)
+    if("clang" IN_LIST FEATURES)
+        file(GLOB_RECURSE _clang_debug_targets
+            "${CURRENT_PACKAGES_DIR}/share/clang/*-release.cmake"
+        )
+    endif()
+    foreach(_target IN LISTS _llvm_debug_targets _clang_debug_targets)
         file(READ ${_target} _contents)
         # LLVM tools should be located in the bin folder because llvm-config expects to be inside a bin dir.
         # Rename `/tools/${PORT}` to `/bin` back because there is no way to avoid this in vcpkg_fixup_cmake_targets.

--- a/ports/llvm/portfile.cmake
+++ b/ports/llvm/portfile.cmake
@@ -169,6 +169,9 @@ vcpkg_configure_cmake(
         -DLLVM_BUILD_EXAMPLES=OFF
         -DLLVM_INCLUDE_TESTS=OFF
         -DLLVM_BUILD_TESTS=OFF
+        # Disable optional dependencies to libxml2 and zlib
+        -DLLVM_ENABLE_LIBXML2=OFF
+        -DLLVM_ENABLE_ZLIB=OFF
         # Force TableGen to be built with optimization. This will significantly improve build time.
         -DLLVM_OPTIMIZED_TABLEGEN=ON
         # LLVM generates CMake error due to Visual Studio version 16.4 is known to miscompile part of LLVM.

--- a/ports/llvm/portfile.cmake
+++ b/ports/llvm/portfile.cmake
@@ -215,12 +215,11 @@ endif()
 if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
     file(GLOB_RECURSE _llvm_debug_targets
         "${CURRENT_PACKAGES_DIR}/share/llvm/*-debug.cmake"
-        "${CURRENT_PACKAGES_DIR}/share/clang/*-debug.cmake"
     )
     set(_clang_debug_targets)
     if("clang" IN_LIST FEATURES)
         file(GLOB_RECURSE _clang_debug_targets
-            "${CURRENT_PACKAGES_DIR}/share/clang/*-release.cmake"
+            "${CURRENT_PACKAGES_DIR}/share/clang/*-debug.cmake"
         )
     endif()
     foreach(_target IN LISTS _llvm_debug_targets _clang_debug_targets)

